python/analysis/raster/qgsalignraster.sip

```diff
@@ -1,183 +1,325 @@
+/************************************************************************
+ * This file has been generated automatically from                      *
+ *                                                                      *
+ * src/analysis/raster/qgsalignraster.h                                 *
+ *                                                                      *
+ * Do not edit manually ! Edit header and run scripts/sipify.pl again   *
+ ************************************************************************/
+
+
+
+
+typedef void *GDALDatasetH;
+
 
 class QgsAlignRaster
 {
-%TypeHeaderCode
-#include <qgsalignraster.h>
-#include <gdal_version.h>
+%Docstring
+ QgsAlignRaster takes one or more raster layers and warps (resamples) them
+ so they have the same:
+ - coordinate reference system
+ - cell size and raster size
+ - offset of the raster grid
+
+.. versionadded:: 2.12
 %End
 
+%TypeHeaderCode
+#include "qgsalignraster.h"
+%End
   public:
     QgsAlignRaster();
 
-    //! Utility class for gathering information about rasters
     struct RasterInfo
     {
-    public:
-      //! Construct raster info with a path to a raster file
-      RasterInfo( const QString &layerpath );
-      ~RasterInfo();
-
-      //! Check whether the given path is a valid raster
-      bool isValid() const;
-
-      //! Return CRS in WKT format
-      QString crs() const;
-      //! Return size of the raster grid in pixels
-      QSize rasterSize() const;
-      //! Return number of raster bands in the file
-      int bandCount() const;
-      //! Return cell size in map units
-      QSizeF cellSize() const;
-      //! Return grid offset
-      QPointF gridOffset() const;
-      //! Return extent of the raster
-      QgsRectangle extent() const;
-      //! Return origin of the raster
-      QPointF origin() const;
-
-      //! write contents of the object to standard error stream - for debugging
-      void dump() const;
-
-      //! Get raster value at the given coordinates (from the first band)
-      double identify( double mx, double my );
-
-      private:
-        RasterInfo( const QgsAlignRaster::RasterInfo &rh );
```

Comment:
`SIP_FORCE` if the method should be there both in C++ and in SIP, `#ifdef SIP_RUN` if only in the SIP file.
Since it is not present in current header at the moment, it requires a  `#ifdef SIP_RUN`.